src/materialized/bin/materialized.rs

```diff
@@ -161,6 +161,7 @@ fn run() -> Result<(), failure::Error> {
     };
 
     let data_directory = popts.opt_get_default("data-directory", PathBuf::from("mzdata"))?;
+    fs::create_dir_all(data_directory.clone())?;
```

Comment:
its a mystery to me why I need this clone when this function call didn't need it in the previous invocation in clone. compiler seems to think the value is moved and then later borrowed further down this function but its not clear to me how thats different from what was happening in the coord crate earlier

Reply:
Guessing, but: the argument to the method needs to implement `AsRef<Path>`, and you might have gotten yourself a reference before (which is `Copy`) and an owned thing this time. Does it work with a `&` in front instead of the `clone()`?

Reply:
yes thats right. ok now I'm relooking through the old stuff and the data_directory in the coord config is indeed a ref to path

Reply:
and here is where we turned it into a ref
```
255             data_directory: config.data_directory.as_deref(),
```
checks out